csharp/ql/lib/semmle/code/csharp/controlflow/Guards.qll

```diff
@@ -323,229 +321,39 @@ class Guard extends Expr {
    */
   predicate controlsBasicBlock(BasicBlock bb, AbstractValue v) { guardControls(this, bb, v) }
 
-  /**
-   * Holds if this guard is an equality test between `e1` and `e2`. If the test is
-   * negated, that is `!=`, then `polarity` is false, otherwise `polarity` is
-   * true.
-   */
-  predicate isEquality(Expr e1, Expr e2, boolean polarity) {
-    exists(BooleanValue v |
-      this = getAnEqualityCheck(e1, v, e2) and
-      polarity = v.getValue()
-    )
-  }
-
   /**
    * Gets a valid value for this guard. For example, if this guard is a test, then
    * it can have Boolean values `true` and `false`.
    */
-  AbstractValue getAValue() { isGuard(this, result) }
+  deprecated AbstractValue getAValue() { isGuard(this, result) }
 }
 
-/** An abstract value. */
-abstract class AbstractValue extends TAbstractValue {
-  /** Holds if the `s` branch out of `cfe` is taken iff `e` has this value. */
-  abstract predicate branch(ControlFlowElement cfe, ConditionalSuccessor s, Expr e);
-
-  /** Gets an abstract value that represents the dual of this value, if any. */
-  abstract AbstractValue getDualValue();
-
-  /**
-   * Gets an expression that has this abstract value. Two expressions that have the
-   * same concrete value also have the same abstract value, but not necessarily the
-   * other way around.
-   *
-   * Moreover, `e = this.getAnExpr() implies not e = this.getDualValue().getAnExpr()`.
-   */
-  abstract Expr getAnExpr();
-
-  /**
-   * Holds if this is a singleton abstract value. That is, two expressions that have
-   * this abstract value also have the same concrete value.
-   */
-  abstract predicate isSingleton();
-
-  /**
-   * Holds if this value describes a referential property. For example, emptiness
-   * of a collection is a referential property.
-   *
-   * Such values only propagate through adjacent reads, for example, in
-   *
-   * ```csharp
-   * int M()
-   * {
-   *     var x = new string[]{ "a", "b", "c" }.ToList();
-   *     x.Clear();
-   *     return x.Count;
-   * }
-   * ```
-   *
-   * the non-emptiness of `new string[]{ "a", "b", "c" }.ToList()` only propagates
-   * to the read of `x` in `x.Clear()` and not in `x.Count`.
-   *
-   * Aliasing is not taken into account in the analyses.
-   */
-  predicate isReferentialProperty() { none() }
-
-  /** Gets a textual representation of this abstract value. */
-  abstract string toString();
-}
+class AbstractValue = GuardValue;
```

Comment:
deprecated?

Reply:
I figured I'd do that in a follow-up PR.